### PR TITLE
Update JS documentation for international pages

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Update JS documentation for international pages
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no need
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24879)

### CODE REVIEW

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 1]
```diff
@@ -164,10 +173,10 @@ class Countries extends BOBasePage {
 
   /**
    * Filter table
-   * @param page
-   * @param filterType
-   * @param filterBy
-   * @param value
+   * @param page {Page} Browser tab
+   * @param filterType {string} Input or select to choose method of filter
+   * @param filterBy {string} Column to filter}
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
   * @param filterBy {string} Column to filter
```

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/international/localization/localizationBasePage.js [hunk 2]
```diff
@@ -22,7 +31,7 @@ module.exports = class LocalizationBasePage extends BOBasePage {
 
   /**
    * Go to currencies page
-   * @param page
+   * @param page Browser tab
```
Comment (Progi1984) [inline tests/UI/pages/BO/international/localization/localizationBasePage.js]:
```suggestion
   * @param page {Page} Browser tab
```

File: tests/UI/pages/BO/international/localization/localizationBasePage.js [hunk 3]
```diff
@@ -13,7 +22,7 @@ module.exports = class LocalizationBasePage extends BOBasePage {
   /* Header Methods */
   /**
    * Go to languages page
-   * @param page
+   * @param page Browser tab
```
Comment (Progi1984) [inline tests/UI/pages/BO/international/localization/localizationBasePage.js]:
```suggestion
   * @param page {Page} Browser tab
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

File: tests/UI/pages/BO/international/localization/index.js [hunk 4]
```diff
@@ -30,10 +40,10 @@ class Localization extends LocalizationBasePage {
   /* Methods */
   /**
    * Import a localization pack
-   * @param page
-   * @param country
-   * @param contentToImport
-   * @param downloadPackData
+   * @param page {Page} Browser tab
+   * @param country {string} Country to select
+   * @param contentToImport {contentToImport} Data of content to import to choose
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/localization/index.js:45]: What is a contenToImport type?

If it's an object, document it as `{Object}`, same for others types such as zoneData

```suggestion
   * @param contentToImport {Object} Data of content to import to choose
```

File: tests/UI/pages/BO/international/locations/add.js [hunk 5]
```diff
@@ -19,8 +28,8 @@ class AddZone extends BOBasePage {
    */
   /**
    * Fill form for add/edit zone
-   * @param page
-   * @param zoneData
+   * @param page {Page} Browser tab
+   * @param zoneData {zoneData} Data to set on new/edit zone page
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/add.js:32]:
```suggestion
   * @param zoneData {Object} Data to set on new/edit zone page
```

File: tests/UI/pages/BO/international/locations/countries/add.js [hunk 6]
```diff
@@ -28,8 +37,8 @@ class AddCountry extends BOBasePage {
    */
   /**
    * Fill form for add/edit country
-   * @param page
-   * @param countryData
+   * @param page {Page} Browser tab
+   * @param countryData {countryData} Data to set on new country form
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/add.js:41]:
```suggestion
   * @param countryData {Object} Data to set on new country form
```

File: tests/UI/pages/BO/international/locations/states/add.js [hunk 7]
```diff
@@ -22,8 +31,8 @@ class AddState extends BOBasePage {
    */
   /**
    * Fill form for add/edit state
-   * @param page
-   * @param stateData
+   * @param page {Page} Browser tab
+   * @param stateData {stateDate} Data to set on new/edit state form
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/states/add.js:35]:
```suggestion
   * @param stateData {Object} Data to set on new/edit state form
```

File: tests/UI/pages/BO/international/taxes/taxRules/add.js [hunk 8]
```diff
@@ -32,22 +42,23 @@ class AddTaxRules extends BOBasePage {
...
+
     return this.getAlertSuccessBlockContent(page);
   }
 
   /**
    * Fill form for add/edit tax rules group
-   * @param page
-   * @param taxRuleData
+   * @param page {Page} Browser tab
+   * @param taxRuleData {taxRuleData} Data to set on new/edit tax rule data
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/taxes/taxRules/add.js:61]:
```suggestion
   * @param taxRuleData {Object} Data to set on new/edit tax rule data
```

File: tests/UI/pages/BO/international/taxes/taxRules/add.js [hunk 9]
```diff
@@ -32,22 +42,23 @@ class AddTaxRules extends BOBasePage {
 
   /**
    * Fill form for add/edit tax rules group
-   * @param page
-   * @param taxRuleGroupData
+   * @param page {Page} Browser tab
+   * @param taxRuleGroupData {taxRuleGroupData} Data to set on tax rule group data
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/taxes/taxRules/add.js:46]:
```suggestion
   * @param taxRuleGroupData {Object} Data to set on tax rule group data
```

**Comment by NeOMakinG** [comment]
Is `Page` the real type name? Isn't it `BOBasePage` or something else?

**Comment by boubkerbribri** [comment]
@NeOMakinG About your comments :
> For `Object` instead of `somethingData`: 

This update is now done after a comment that you did to stick with data type instead of Object that could mean anything. So now, we are adding types on data files too and changes the type in pages to `somethingData`

> Is Page the real type name? Isn't it BOBasePage or something else?

No, it's a class from playwright that means 'A browser tab' and `BOBasePage` is something different that, it's a Page Object from the POM architecture.

**Comment by NeOMakinG** [comment]
> @NeOMakinG About your comments :
> 
> > For `Object` instead of `somethingData`:
> 
> This update is now done after a comment that you did to stick with data type instead of Object that could mean anything. So now, we are adding types on data files too and changes the type in pages to `somethingData`
> 

Oh ok, so it's a real type, interfaced by a JSON object 👍

**Review by NeOMakinG — DISMISSED** [review]

File: tests/UI/pages/BO/international/localization/localizationBasePage.js (hunk 2 above)
Comment (boubkerbribri) [inline tests/UI/pages/BO/international/localization/localizationBasePage.js]:
```suggestion
   * @param page {Page} Browser tab
```

**Review by boubkerbribri — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**